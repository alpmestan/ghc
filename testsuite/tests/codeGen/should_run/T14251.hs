{-# LANGUAGE MagicHash, BangPatterns #-}
module Main where

-- A minor modification from T8064.hs.
--
-- The key here is that we ensure that
-- subsequently passed floats do not
-- accidentally end up in previous
-- registers.
--

import GHC.Exts

{-# NOINLINE f #-}
f :: (Int# -> Float# -> Double# -> Float# -> Double# -> String) -> String
f g = g 3# 4.0# 5.0## 6.0# 6.9## ++ " World!"

{-# NOINLINE q #-}
q :: Int# -> Float# -> Double# -> Float# -> Double# -> String
q i j k l m = "Hello " ++ show (F# l) ++ " " ++ show (D# m)

main = putStrLn (f $ q)
