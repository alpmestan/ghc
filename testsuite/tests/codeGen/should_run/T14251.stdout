Hello 6.0 6.9 World!
